packages/storybook/src/nimble/chip/chip-matrix.stories.ts

```diff
@@ -0,0 +1,127 @@
+import type { StoryFn, Meta } from '@storybook/html-vite';
+import { html, ViewTemplate, when } from '@ni/fast-element';
+import { iconKeyTag } from '@ni/nimble-components/dist/esm/icons/key';
+import { ChipAppearance } from '@ni/nimble-components/dist/esm/chip/types';
+import { chipTag } from '@ni/nimble-components/dist/esm/chip';
+import {
+    controlLabelFont,
+    controlLabelFontColor
+} from '@ni/nimble-components/dist/esm/theme-provider/design-tokens';
+import {
+    createMatrix,
+    sharedMatrixParameters,
+    createMatrixThemeStory,
+    cartesianProduct,
+    createMatrixInteractionsFromStates
+} from '../../utilities/matrix';
+import { disabledStates, type DisabledState } from '../../utilities/states';
+import { textCustomizationWrapper } from '../../utilities/text-customization';
+
+const appearanceStates = [
+    ['Outline', ChipAppearance.outline],
+    ['Block', ChipAppearance.block]
+] as const;
+type AppearanceState = (typeof appearanceStates)[number];
+
+const removableStates = [
+    ['Removable', true],
+    ['Not Removable', false]
+] as const;
+type RemovableStates = (typeof removableStates)[number];
+
+const showStartSlotIconStates = [
+    ['Show Icon', true],
+    ['Hide Icon', false]
+] as const;
+type ShowStartSlotIconState = (typeof showStartSlotIconStates)[number];
+const showStartSlotIconStatesOnlyIcon = showStartSlotIconStates[0];
+
+const labelStates = [
+    ['Short Label', 'Option Label'],
+    [
+        'Long Label',
+        'A very long label that will likely overflow the chip and require truncation with set width'
+    ]
+] as const;
+type LabelState = (typeof labelStates)[number];
+const labelStatesOnlyShort = labelStates[0];
+
+const widthStates = [
+    ['Default Width', ''],
+    ['Custom Width', 'width: 250px;']
```

Comment:
If you haven't already, can you confirm with @NIbokeefe that when the chip is wider than the default "fit-text" width, that the remove button should stick to the text rather than the edge of the chip?

<img width="299" height="61" alt="image" src="https://github.com/user-attachments/assets/32ae94fe-37bc-4750-89eb-458446334b77" />

FWIW - in my opinion, the remove button should always be right-aligned in the chip rather than floating in the middle next text. I also wonder if the text & icon should always be left aligned rather than centered.

Reply:
@atmgrifter00 @mollykreis,

I don't see the use case for wanting to expand a chip's width.  We do allow that with buttons. However, with chips anywhere in the short or long term they should be nested within a control and not a stand alone selection element that would require modified margins to fill spaces.

If it makes sense to build that functionality into the control regardless, I agree with @mollykreis, the text and icon should left align, and the close button should right align as shown in example "A."  We should prevent this from happening until we have a use case that warrants it.

<img width="717" height="494" alt="Screenshot 2025-08-25 at 12 11 07 PM" src="https://github.com/user-attachments/assets/40bd23ce-b0b8-418b-bd70-f1f8be4da5c2" />